pandas/tests/io/test_clipboard.py

```diff
@@ -243,6 +243,52 @@ def test_read_clipboard_infer_excel(self, request, mock_clipboard):
 
         tm.assert_frame_equal(res, exp)
 
+    def test_infer_excel_with_nulls(self, request, mock_clipboard):
+        # GH41108
+        text = "col1\tcol2\n1\tred\n\tblue\n2\tgreen"
+
+        mock_clipboard[request.node.name] = text
+        df = read_clipboard()
+        df_expected = DataFrame(
+            data={"col1": [1, None, 2], "col2": ["red", "blue", "green"]}
+        )
+
+        # excel data is parsed correctly
+        tm.assert_frame_equal(df, df_expected)
+
+    @pytest.mark.parametrize(
+        "multiindex",
+        [
+            (
+                """\t\t\tcol1\tcol2
+                A\t0\tTrue\t1\tred
+                A\t1\tTrue\t\tblue
+                B\t0\tFalse\t2\tgreen""",
+                [["A", "A", "B"], [0, 1, 0], [True, True, False]],
+            ),
+            (
+                """\t\tcol1\tcol2
+                A\t0\t1\tred
+                A\t1\t\tblue
+                B\t0\t2\tgreen""",
+                [["A", "A", "B"], [0, 1, 0]],
+            ),
+        ],
+    )
+    def test_infer_excel_with_multiindex(self, request, mock_clipboard, multiindex):
+        # GH41108
+
+        # the `.replace()` is because `.dedent()` does not like the leading `\t`
+        mock_clipboard[request.node.name] = multiindex[0].replace(" ", "")
```

Comment:
Can you explain this more? Where is dedent being used?

Reply:
To make it a bit more readable I changed the text used to mock the clipboard from
```python
text = "\t\tcol1\tcol2\nA\t0\t1\tred\nA\t1\t\tblue\nB\t0\t2\tgreen"
```
to 
```python
               """\t\t\tcol1\tcol2
                A\t0\tTrue\t1\tred
                A\t1\tTrue\t\tblue
                B\t0\tFalse\t2\tgreen"""
```

That's similar to how it's done in some of the other tests, except now there are extra spaces/tabs because of the multiline string.   
The other tests use `.dedent()` to get rid of it, but in this case it is also deleting the leading tabs for the index which is what we are testing and i couldn't get it to work properly, so I'm instead removing the spaces directly

Reply:
That makes sense - so maybe change the comment to something more like "Can't use dedent as it will remove the leading \t" and move to the line where this string is set above.

As of now, the comment reads to me something like "we use replace because if we didn't, when we use dedent later things will go wrong".

Reply:
I think you are right and the replace was a bit weird to begin with, i've changed it now to concatenate each line, i hope that's better